### PR TITLE
Docs: Document `windowHistorySupport` flag, and add `pushState`/`replaceState` examples

### PR DESCRIPTION
**Pull request by delbaoliveira**
Closes: https://linear.app/vercel/issue/DX-2273/document-historypushstate-and-historyreplacestate

### CODE REVIEW

File: docs/02-app/01-building-your-application/01-routing/03-linking-and-navigating.mdx [hunk 1]
```diff
@@ -183,6 +183,70 @@ For a full list of `useRouter` methods, see the [API reference](/docs/app/api-re
...
+```tsx fileName="app/ui/sort-products.tsx"
+'use client'
+
+import { useCallback } from 'react'
+import { useSearchParams } from 'next/navigation'
+
+export default function SortProducts() {
+  const searchParams = useSearchParams()
+
+  const updateSorting = useCallback(
```
Comment (leerob) [inline docs/02-app/01-building-your-application/01-routing/03-linking-and-navigating.mdx:205]: Do we need `useCallback`?

Comment (leerob) [reply docs/02-app/01-building-your-application/01-routing/03-linking-and-navigating.mdx:205]: Could we add JS code snippets here too?

File: docs/02-app/01-building-your-application/01-routing/03-linking-and-navigating.mdx [hunk 2]
```diff
@@ -183,6 +183,70 @@ For a full list of `useRouter` methods, see the [API reference](/docs/app/api-re
...
+    </>
+  )
+}
+```
+
+### `window.history.replaceState`
+
+Use it to replace the current entry on the browser's history stack. The user is not able to navigate back to the previous state. For example, to switch the application's locale:
+
+```tsx fileName="app/ui/locale-switcher.tsx"
```
Comment (leerob) [inline docs/02-app/01-building-your-application/01-routing/03-linking-and-navigating.mdx:227]: Could we add JS code snippets here too?

File: docs/02-app/01-building-your-application/01-routing/03-linking-and-navigating.mdx [hunk 3]
```diff
@@ -183,6 +183,70 @@ For a full list of `useRouter` methods, see the [API reference](/docs/app/api-re
...
+'use client'
+
+import { usePathname } from 'next/navigation'
+
+export default function LocaleSwitcher() {
```
Comment (leerob) [inline docs/02-app/01-building-your-application/01-routing/03-linking-and-navigating.mdx:232]: Suuuuuper small nits. Prefer named exports for non-pages. Prefer `function switchLocale` versus the anon function.

File: docs/02-app/02-api-reference/05-next-config-js/windowHistorySupport.mdx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+---
+title: webVitalsAttribution
```
Comment (delbaoliveira) [inline docs/02-app/02-api-reference/05-next-config-js/windowHistorySupport.mdx:2]:
```suggestion
title: windowHistorySupport
description: Learn how to use the windowHistorySupport to sync the History API with the Next.js Router. 
```

File: docs/02-app/02-api-reference/05-next-config-js/windowHistorySupport.mdx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+description: Learn how to use the webVitalsAttribution option to pinpoint the source of Web Vitals issues.
```
Comment (delbaoliveira) [inline docs/02-app/02-api-reference/05-next-config-js/windowHistorySupport.mdx:3]:
```suggestion
title: windowHistorySupport
description: Learn how to use the windowHistorySupport to sync the History API with the Next.js Router. 
```

**Review by leerob — APPROVED** [review]

File: docs/02-app/02-api-reference/05-next-config-js/windowHistorySupport.mdx (hunk 5 above)
Comment (A7med3bdulBaset) [reply docs/02-app/02-api-reference/05-next-config-js/windowHistorySupport.mdx:3]: You probably forgot to commit this change

I made a follow up #60503

Comment (delbaoliveira) [reply docs/02-app/02-api-reference/05-next-config-js/windowHistorySupport.mdx:3]: Thank you 🙏🏼

**Pull request merged**